### PR TITLE
d/jak2: finish `cylinder` | `mech-h` | `font-data` most of `text` and `history` and chip away at a bunch of other files

### PR DESCRIPTION
**Pull request by xTVaser**
- `level-info` has an issue with creating arrays of symbols from static data
- `text` has a CFG failure and the `print..` function seems to do something different, not sure if we manually fixed this one as it was never tested in jak 1's ref tests

### CODE REVIEW

File: goal_src/jak2/engine/debug/history.gc [hunk 1]
```diff
@@ -5,5 +5,587 @@
...
+  (format #t "~1Torigin: #<vector @ #x~X>~%" (-> obj origin))
+  (format #t "~1Tbytes[16] @ #x~X~%" (-> obj bytes))
+  (format #t "~1Tvector: #<vector @ #x~X>~%" (-> obj bytes))
+  (format #t "~1Tfloat: ~f~%" (-> obj vector x))
+  (format #t "~1Tcollide-status: ~D~%" (-> obj collide-status))
+  (format #t "~1Tcollide-reaction-flag: ~D~%" (-> obj vector z))
+  (format #t "~1Tpat: ~D~%" (-> obj vector x))
+  (label cfg-4)
+  obj
+  )
```
Comment (water111) [inline goal_src/jak2/engine/debug/history.gc:117]: you could remove the inspects here if you want.

File: goal_src/jak2/engine/debug/history.gc [hunk 2]
```diff
@@ -5,5 +5,587 @@
...
 
+(defmethod clear-history-entries! history ((obj history))
+  "Iterates through each [[history-elt]] in the `elt` dynamic array
+       For each entry:
+       - clear `timestamp`
+       - clear `record-tag`"
+  (set! (-> obj alloc-index) 0)
+  (countdown (v1-0 (-> obj allocated-length))
+    (let ((a1-3 (the-as history-elt (+ (+ (* 48 v1-0) 12) (the-as int obj)))))
```
Comment (water111) [inline goal_src/jak2/engine/debug/history.gc:206]: Hm, this probably should be
```
(-> obj elts v1-0)
```
In the `history` deftype, it should probably be
```
(elts history-elt :dynamic :inline :offset 16)
```
might need a `:pack-me` on the `history-elt`, not sure.

File: goal_src/jak2/engine/debug/history.gc [hunk 3]
```diff
@@ -5,5 +5,587 @@
...
+    )
+  0
+  (none)
+  )
+
+(define-perm *history* history (new 'debug 'history 4096))
+
+(clear-history-entries! *history*)
+
+;; ERROR: function was not converted to expressions. Cannot decompile.
```
Comment (water111) [inline goal_src/jak2/engine/debug/history.gc:220]: Looks like this one works if you remove it from `hacks.json`
Maybe
```
(history-method-asm-9 (_type_ int int int) object 9)
```

(with one of those ints being `history-channel`)

File: goal_src/jak2/engine/debug/history.gc [hunk 4]
```diff
@@ -5,5 +5,587 @@
...
+    (while (not (-> obj done?))
+      (let ((a2-1 (+ (-> obj index) -1)))
+        (if (< a2-1 0)
+            (set! a2-1 (+ (-> v1-0 allocated-length) -1))
+            )
+        (set! (-> obj index) a2-1)
+        (if (= a2-1 (-> v1-0 alloc-index))
+            (set! (-> obj done?) #t)
+            )
+        (let ((a2-5 (the-as history-elt (+ (+ (* 48 a2-1) 12) (the-as int v1-0)))))
```
Comment (water111) [inline goal_src/jak2/engine/debug/history.gc:252]: this also should hopefully get cleaned up by making the `history-elt`s inline.

File: goal_src/jak2/engine/level/level-h.gc [hunk 5]
```diff
@@ -287,7 +290,7 @@
     (debug-print-region-splitbox (_type_ vector object) none 20)
     (get-art-group-by-name (_type_ string) art-group 21)
     (level-method-22 () none 22)
-    (level-method-23 () none 23)
+    (level-method-23 (_type_ game-text-id symbol) string 23)
```
Comment (water111) [inline goal_src/jak2/engine/level/level-h.gc:293]: Probably `lookup-text` or something like that.

**Review by water111 — CHANGES_REQUESTED** [review]

File: decompiler/config/jak2/anonymous_function_types.jsonc [hunk 6]
```diff
@@ -7,8 +7,6 @@
     [23, "(function process symbol)"],
     [17, "(function process symbol)"]
   ],
-  "level": [[5, "(function none)"]],
```
[the remark this answers is not in the document]
Comment (xTVaser) [reply decompiler/config/jak2/anonymous_function_types.jsonc:10]: cleaning up duplicate json keys

**Review by water111 — APPROVED** [review]

**Pull request merged**